modules/dashboard/php/ajaxdashboard.class.inc

```diff
@@ -0,0 +1,547 @@
+use \Psr\Http\Message\ResponseInterface;
+
+/**
+ * The NDB_Page for the POST requests.
+ *
+ * @category Loris
+ * @package  Dashboard
+ * @author   Tara Campbell <tara.campbell@mail.mcgill.ca>
+ *           Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class AjaxDashboard extends \NDB_Page
+{
+    public $skipTemplate = true;
+    /**
+     * This function will return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        /**
+         * Ensure it's a POST request.
+         */
+        if ($request->getMethod() != 'POST') {
+            return (new \LORIS\Http\Response())
+                ->withHeader('Content-Type', 'application/json')
+                ->withStatus(405)
+                ->withHeader('Allow', 'POST')
+                ->withBody(
+                    new \LORIS\Http\StringStream(
+                        'Only POST requests are allowed'
+                    )
+                );
+        }
+        /**
+         * Parse POST request body.
+         */
+        $requestBody = $request->getParsedBody();
+
+        /**
+         * Response to reply.
+         */
+        $response = $this->processRequest($requestBody);
+        return (new \LORIS\Http\Response())
+            ->withHeader('Content-Type', 'application/json')
+            ->withStatus(200)
+            ->withHeader('Allow', 'POST')
+            ->withBody(
+                new \LORIS\Http\StringStream(
+                    json_encode($response)
+                )
+            );
+    }
+    /**
+     * Processes the values and saves to database
+     *
+     * @param array $values the user data received.
+     *
+     * @return array
+     */
+    function processRequest(array $values) : array
+    {
+        $response = array();
+        if (isset($values['command']) && $values['command'] == 'initialize') {
+            $response = $this->initializeSetup($values);
+        } else {
+            $response['status'] = 'error';
+        }
+        return $response;
+    }
+
+    /**
+     * Initialize setup, building the Dashboard.
+     *
+     * @param array $values the user data received.
+     *
+     * @return array
+     * @throws \LorisException
+     */
+    function initializeSetup(array $values) : array
+    {
+        // data for panels.
+        $welcomeData            = array();
+        $recruitmentData        = array();
+        $studyProgressionData   = array();
+        $myTasksData            = array();
+        $documentRepositoryData = array();
+        $behaviouralFeedbackData = array();
+
+        $DB     = \Database::singleton();
+        $user   = \User::singleton();
```

Comment:
$user = $factory->user();

Side note: it would be easier if this function was receiving the $request and returning a ResponseInterface because the user could be retrieved from the $request using getAttribute('user'); to avoid a static call. But since there is already static calls here, let's revisit this later (: